api/Updated-Metrics.md

```diff
@@ -0,0 +1,406 @@
+```python
+>>> difference_from_summary(result)
+0.4406
+>>> ratio_from_summary(result)
+0.3259
+>>> group_max_from_summary(result)
+0.6536
+>>> group_min_from_summary(result)
+0.2130
+```
+We also provide wrappers such as `accuracy_score_difference()`, `accuracy_score_ratio()` and `accuracy_score_min()` for user convenience.
+
+### Proposed Change
+
+Although the functionality of the `group_max_from_summary()` and `group_min_from_summary()` can be accessed by calling `metrics.by_group.min()` and `metrics.by_group.max()`, we will provide `.group_min()` and `.group_max()` methods for completeness.
+
+For `difference_from_summary()` and `ratio_from_summary()` we will add methods to calculate the values as we do now, and also relative to the `overall` value.
+First for computing the difference:
+```python
+>>> metrics.difference()
+accuracy_score 0.4406
+dtype: float64
+>>> metrics.difference_to_overall()
+accuracy_score 0.2563 # max(abs(0.6536-0.4), abs(0.213-0.4))
+dtype: float64
+```
+Note that the result type is a DataFrame (for reasons which will become clear below).
+
+We would similarly have `ratio()` and `ratio_to_overall()`methods:
+```python
+>>> metrics.ratio()
+accuracy_score 0.3259
+dtype: float64
+>>> metrics.ratio_to_overall()
+accuracy_score 0.6120 # min(abs(0.4/0.6536), abs(0.213/0.6536))
+```
+
+## Intersections of Sensitive Features
+
+### Existing Syntax
+
+Our current API does not support evaluating metrics on intersections of sensitive features (e.g. "black and female", "black and male", "white and female", "white and male").
+To achieve this, users currently need to write something along the lines of:
+```python
+>>> A_combined = A['SF 1'] + '-' + A['SF 2']
+
+>>> accuracy_score_group_summary(y_true, y_pred,
+                                 sensitive_features=A_combined)
+{ 'overall': 0.4, by_groups : { 'B-M':0.4, 'B-N':0.5, 'B-P':0.5, 'C-M':0.5, 'C-N': 0.6, 'C-P':0.7 } }
+```
+This is unecessarily cumbersome.
+It is also possible that some combinations might not appear in the data (especially as more sensitive features are combined), but identifying which ones were not represented in the dataset would be tedious.
+
+
+### Proposed Change
+
+If `sensitive_features=` is a DataFrame (or list of Series.... exact supported types are TBD), we can generate our results in terms of a MultiIndex. Using the `A` DataFrame defined above, a user might write:
+```python
+>>> result = group_summary(skm.accuracy_score,
+                           y_true, y_pred,
+                           sensitive_features=A)
+>>> result.by_groups
+           accuracy_score
+SF 1 SF 2
+B    M               0.50
+     N               0.40
+     P               0.55
+C    M               0.45
+     N               0.70
+     P               0.63
+```
+If a particular combination of sensitive features had no representatives, then we would return `None` for that entry in the Series.
```

Comment:
Why do we need to return anything? Multi-index does not require the cartesian product structure, if there's no examples with `B-N` combination, we could just return:
```python
>>> result.by_groups
           accuracy_score
SF 1 SF 2
B    M               0.50
     P               0.55
C    M               0.45
     N               0.70
     P               0.63
```

Reply:
That's OK if a user just iterates over a MultiIndex. What happens if a user goes looking for a particular combination which isn't there? Would they rather get an `IndexOutOfRange` exception (or whatever Pandas does), or get `None` as the result? We can do either.